File: infrastructures/ddd/python/src/infrastructure/persistence/write/infra/base_write_entity.py
"""
Base class for all write entities (command side of CQRS).
Write entities are used when business logic executes against data.
They contain all fields needed for business operations.
"""

from infrastructure.persistence.infra import BaseEntity


class BaseWriteEntity(BaseEntity):
    """
    Base class for write entities (command side).
    
    Write entities:
    - Live in /infrastructure/persistence/write/
    - Have WriteEntity suffix
    - Used for commands (create, update, delete)
    - Contain all fields needed for business operations
    - May have optimistic concurrency control (version field)
    """
    
    def __init__(self):
        """Initialize a new write entity."""
        super().__init__()
        self.version: int = 0  # For optimistic concurrency control

File: infrastructures/ddd/typescript/src/infrastructure/persistence/write/infra/BaseWriteEntity.ts
import { BaseEntity } from '../../infra/BaseEntity';

/**
 * Base class for all Write Entities (command side of CQRS).
 * This class:
 * - Lives in /infrastructure/persistence/write/infra/
 * - Used for business logic execution (commands: create, update, delete)
 * - Contains all fields needed for business operations
 * - CAN have database/ORM attributes
 * - Includes Version for optimistic concurrency
 * 
 * WriteEntities:
 * - Suffix with "WriteEntity"
 * - Live in /infrastructure/persistence/write/
 * - Used when business logic executes against data
 * - Mapped from/to BMO by repository using mapper
 */
export abstract class BaseWriteEntity extends BaseEntity {
  public Version?: number;

  constructor() {
    super();
    this.Version = 1;
  }
}
